### PR TITLE
Consent section: avoid headers without body

### PR DESCRIPTION
**Pull request by p2**
With this change, `ORKConsentSectionFormatter` will return an empty string if a section does not have content.

In our consenting we have 2 or 3 steps that do not have a more info text. These would show up as only a header in the generated HTML.

### CODE REVIEW

File: ResearchKit/Consent/ORKConsentSectionFormatter.m [hunk 1]
```diff
@@ -35,9 +35,12 @@ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 @implementation ORKConsentSectionFormatter
 
 - (NSString *)HTMLForSection:(ORKConsentSection *)section {
-    NSString *title = [NSString stringWithFormat:@"<h4>%@</h4>", section.formalTitle?:(section.title?:@"")];
-    NSString *content = [NSString stringWithFormat:@"<p>%@</p>", section.htmlContent?:(section.escapedContent?:@"")];
-    return [NSString stringWithFormat:@"%@%@", title, content];
+	NSString *content = section.htmlContent ?: (section.escapedContent ?: @"");
+	if (0 == content.length) {
+		return @"";
```
Comment (YuanZhu-apple) [inline ResearchKit/Consent/ORKConsentSectionFormatter.m]: As I understand:
```
NSString *content = section.htmlContent ?: (section.escapedContent ?: @"");
```
`content` cannot be nil, then why need the `if` block?

**Comment by YuanZhu-apple** [comment]
@p2 
Do you think it is better to add a boolean to indicate the section to be excluded from generated HTML?

**Comment by p2** [comment]
I thought about that, but in our case it works just as well. The steps to exclude, like _Study Dashboard_, explain in one sentence something about the app. No need to put those in the consent, and since there is no “Learn More” it's excluded. Otherwise there would be a heading _Study Dashboard_ and no content, so at least in these cases it works well without bool.

The question arises if you have some sections that you indeed want to exclude. You could make the point to just override the whole htmlContent in that case, but then you lose the ability to add the signature to the generated PDF. Unless that part is changed... :)

**Comment by YuanZhu-apple** [comment]
But I feel some people still want to keep the title itself in the HTML file. 

Imagine people want to exclude a section from HTML, but the section has both non empty title and content. 

Providing a boolean can accommodate different use cases.

**Comment by p2** [comment]
Only the title looks kinda bad (it's the reason I made this change), but I agree with your second point.

**Comment by YuanZhu-apple** [comment]
r=me.

**Comment by YuanZhu-apple** [comment]
@p2 Can you update this patch? I want to merge it.

**Comment by p2** [comment]
Sure; done.

**Comment by YuanZhu-apple** [comment]
Thanks!

**Pull request merged**